### PR TITLE
User can login to admin application

### PR DESCRIPTION
**Pull request by giacoletti**
### This PR contains:
- adds Login interface in home view
- adds signIn with j-tockauth
- adjusts routes to registration form view and dashboard view
- adjusts fixtures for registration and login end2end tests
- adjusts end2end tests

@elvitak @Mljungho @DefyCab @tochman 

[Pivotal Tracker Feature](https://www.pivotaltracker.com/story/show/180975783)

Side note:
Possible bug, reproduced locally, needs testing on the deployed application
Screenshot from 2022-01-23 13-53-18
Basically when I register an account and I try to login in with the same email (which is stored in DB, checked using rails console locally)
Screenshot from 2022-01-23 13-56-08
I get an error in the admin application:
Screenshot from 2022-01-23 13-57-59
Like I'm not able to authenticate even if I have the user created.

### CODE REVIEW

File: cypress/fixtures/access _token.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (maxarvid) [inline cypress/fixtures/access _token.json]: it looks like there is a whitespace in the title of this file

File: cypress/integration/userCanLogIn.feature.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+describe("A journalist is able to login to use the platform", () => {
+  describe('by using valid credentials', () => {
```
Comment (maxarvid) [inline cypress/integration/userCanLogIn.feature.js]: format code

File: cypress/integration/userCanLogIn.feature.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    before(() => {
+      cy.intercept("POST", "api/auth/sign_in", {
+        fixture: "authenticated_journalist_response.json",
+      }).as('authenticateRequest');
```
Comment (maxarvid) [inline cypress/integration/userCanLogIn.feature.js]: format code

File: cypress/integration/userCanLogIn.feature.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        headers: { uid: "johnskoglung@test.com", token: "12344556789" },
+      });
+      cy.visit("/");
+      cy.get("[data-cy=login-email-input]").type("johnskoglung@test.com");
+      cy.get("[data-cy=login-password-input]").type("1234567890");
+      cy.get("[data-cy=login-button]").click();
+    });
+  
+    it('is expected to make an authentication request to API', () => {
+      cy.wait('@authenticateRequest').its('response.statusCode').should('eq', 200);
```
Comment (maxarvid) [inline cypress/integration/userCanLogIn.feature.js]: format code

File: src/components/HomeView.jsx [hunk 5]
```diff
@@ -1,15 +1,67 @@
...
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = async () => {
+    const response = await Authentication.signIn(
+      userData.email,
+      userData.password
+    );
+    if(response.status === 'success'){
```
Comment (maxarvid) [inline src/components/HomeView.jsx]: format code

File: cypress/fixtures/access _token.json (hunk 1 above)
Comment (maxarvid) [reply cypress/fixtures/access _token.json]: Also, do you need this file? Where in this PR is it used?

File: src/components/HomeView.jsx [hunk 6]
```diff
@@ -1,15 +1,67 @@
...
+      dispatch({ type: 'SET_WELCOME_MESSAGE', payload: `Welcome ${response.data.name}!` });
+      dispatch({ type: 'SET_CURRENT_USER', payload: response.data });
```
Comment (maxarvid) [inline src/components/HomeView.jsx]: You are storing the user's name twice: once in `SET_WELCOME_MESSAGE` and once in `SET_CURRENT_USER`. I suggest removing the `SET_WELCOME_MESSAGE` reducer from application state and instead grabbing the user's name from `currentUser` for your dashboard.

File: src/state/reducers/rootReducer.js [hunk 7]
```diff
@@ -1,5 +1,24 @@
...
+      };
+    case 'SET_WELCOME_MESSAGE':
+      return {
+        ...state,
+        welcomeMessage: action.payload
+      };
+
+    default:
+      return state;
+  }
```
Comment (maxarvid) [inline src/state/reducers/rootReducer.js:16]: format code

File: src/state/reducers/rootReducer.js [hunk 8]
```diff
@@ -1,5 +1,24 @@
...
+      return {
+        ...state,
+        errorMessage: action.payload
+      };
+    case 'SET_CURRENT_USER':
+      return {
+        ...state,
+        currentUser: action.payload
+      };
+    case 'SET_WELCOME_MESSAGE':
```
Comment (maxarvid) [inline src/state/reducers/rootReducer.js]: See comment above on removing `SET_WELCOME_MESSAGE`

File: src/modules/Authentication.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
-import auth from '../modules/auth';
+import auth from './auth';
```
Comment (maxarvid) [inline src/modules/Authentication.js]: format code

File: src/components/Dashboard.jsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate } from "react-router-dom";
+
+const Dashboard = () => {
+  const { welcomeMessage } = useSelector(state => state);
```
Comment (maxarvid) [inline src/components/Dashboard.jsx]: See comment below on reducing the reducers (apologies for the pun)

**Review by maxarvid — CHANGES_REQUESTED** [review]
A couple of notes, I would like to have a 2nd look once they are addressed. Will look at the FE/BE issue then

File: src/components/HomeView.jsx [hunk 11]
```diff
@@ -21,15 +21,17 @@ const HomeView = () => {
...
-      dispatch({ type: 'SET_WELCOME_MESSAGE', payload: `Welcome ${response.data.name}!` });
-      dispatch({ type: 'SET_CURRENT_USER', payload: response.data });
-      navigate('dashboard', {
-        state: { flash: `Welcome ${response.data.name}!`, currentUser: response.data },
+    if (response.status === "success") {
+      dispatch({ type: "SET_CURRENT_USER", payload: response.data });
+      navigate("dashboard", {
+        state: {
+          flash: `Welcome ${response.data.name}!`,
+          currentUser: response.data
```
Comment (maxarvid) [inline src/components/HomeView.jsx]: Now that you have redux, I believe relying on this `react-router` hook is duplicating passing the state around, especially since it looks like you are now using the `useSelector` imported from redux in `Dashboard.jsx`. Try to see if you can remove this.

**Review by maxarvid — CHANGES_REQUESTED** [review]

File: src/components/HomeView.jsx (hunk 11 above)
Comment (giacoletti) [reply src/components/HomeView.jsx]: Ops, that's old code, I forgot about it

**Review by maxarvid — APPROVED** [review]
